libnavigation-core/src/main/java/com/mapbox/navigation/core/RoutesCacheClearer.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.utils.DecodeUtils
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.directions.session.RoutesUpdatedResult
+import com.mapbox.navigation.core.preview.RoutesPreviewObserver
+import com.mapbox.navigation.core.preview.RoutesPreviewUpdate
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+internal class RoutesCacheClearer : RoutesObserver, RoutesPreviewObserver {
+
+    private var currentActiveRoutes: List<NavigationRoute> = emptyList()
+    private var currentPreviewsRoutes: List<NavigationRoute>? = null
+
+    override fun onRoutesChanged(result: RoutesUpdatedResult) {
```

Comment:
what if we set two routes (primary + alternative) and then remove alternative by setting only primary route. Does it make sense to clear geometry of alternative from cache? 

current approach makes the SDK works better already, so I don't mind if you decide to do this in a follow up PR

Reply:
> current approach makes the SDK works better already, so I don't mind if you decide to do this in a follow up PR

Actually, it doesn't. It may affect time performance so I'll add these improvements to this PR.

Reply:
Oh, I got you wrong. Then it can be done in a separate PR.